esphome/components/mcp4461/mcp4461.cpp

```diff
@@ -0,0 +1,359 @@
+            terminal_connector = Mcp4461TerminalIdx::MCP4461_TERMINAL_1;
+          }
+          uint8_t new_terminal_value = this->calc_terminal_connector_byte_(terminal_connector);
+          if (new_terminal_value != this->get_terminal_register(terminal_connector)) {
+            ESP_LOGV(TAG, "updating terminal %d to new value %d", (uint8_t) terminal_connector, new_terminal_value);
+            this->set_terminal_register(terminal_connector, new_terminal_value);
+          }
+        }
+      }
+    }
+    this->update_ = false;
+  }
+}
+
+uint16_t Mcp4461Component::get_status_register() {
+  uint8_t reg = 0;
+  reg |= (uint8_t) Mcp4461Addresses::MCP4461_STATUS;
+  reg |= (uint8_t) Mcp4461Commands::READ;
+  uint16_t buf;
+  if (!this->read_byte_16(reg, &buf)) {
+    this->status_set_warning();
+    ESP_LOGW(TAG, "Error fetching status register value");
+    return 0;
+  }
+  return buf;
+}
+
+bool Mcp4461Component::is_writing_() { return (bool) ((this->get_status_register() >> 4) & 0x01); }
+
+uint8_t Mcp4461Component::get_wiper_address_(uint8_t wiper) {
+  uint8_t addr;
+  bool nonvolatile = false;
+  if (wiper > 3) {
+    nonvolatile = true;
+    wiper = wiper - 4;
+  }
+  switch (wiper) {
+    case 0:
+      addr = (uint8_t) Mcp4461Addresses::MCP4461_VW0;
+      break;
+    case 1:
+      addr = (uint8_t) Mcp4461Addresses::MCP4461_VW1;
+      break;
+    case 2:
+      addr = (uint8_t) Mcp4461Addresses::MCP4461_VW2;
+      break;
+    case 3:
+      addr = (uint8_t) Mcp4461Addresses::MCP4461_VW3;
+      break;
+    default:
+      ESP_LOGE(TAG, "unknown wiper specified");
+      return 0;
+  }
+  if (nonvolatile) {
+    addr = addr + 0x20;
+  }
+  return addr;
+}
+
+uint16_t Mcp4461Component::get_wiper_level(uint8_t wiper) {
+  uint8_t reg = 0;
+  uint16_t buf = 0;
+  reg |= this->get_wiper_address_(wiper);
+  reg |= (uint8_t) Mcp4461Commands::READ;
+  if (wiper > 3) {
+    while (this->is_writing_()) {
+      ESP_LOGV(TAG, "delaying during eeprom write");
+    }
+  }
+  if (!this->read_byte_16(reg, &buf)) {
+    this->status_set_warning();
+    if (wiper > 3) {
+      this->status_set_warning();
+      ESP_LOGW(TAG, "Error fetching nonvolatile wiper %d value", wiper);
+    } else {
+      this->status_set_warning();
+      ESP_LOGW(TAG, "Error fetching wiper %d value", wiper);
+    }
+    return 0;
+  }
+  return buf;
+}
+
+void Mcp4461Component::update_wiper_level(uint8_t wiper) {
+  uint16_t data;
+  data = this->get_wiper_level(wiper);
+  ESP_LOGV(TAG, "Got value %d from wiper %d", data, wiper);
+  this->reg_[wiper].state = data;
+}
+
+void Mcp4461Component::set_wiper_level(uint8_t wiper, uint16_t value) {
+  ESP_LOGV(TAG, "Setting MCP4461 wiper %d to %d!", wiper, value);
+  this->reg_[wiper].state = value;
+  this->update_ = true;
+}
+
+void Mcp4461Component::write_wiper_level_(uint8_t wiper, uint16_t value) {
+  if (wiper > 3) {
+    while (this->is_writing_()) {
+      ESP_LOGV(TAG, "delaying during eeprom write");
+    }
```

Comment:
```suggestion
    while (this->is_writing_()) {
      ESP_LOGV(TAG, "delaying during eeprom write");
      yield();
    }
```

Since its used multiple times, move it into function.